monkey/monkey_island/cc/resources/island_mode.py

```diff
@@ -0,0 +1,24 @@
+import json
+import logging
+
+import flask_restful
+from flask import make_response, request
+
+from monkey_island.cc.resources.auth.auth import jwt_required
+from monkey_island.cc.services.mode.island_mode_service import set_mode
+from monkey_island.cc.services.mode.mode_enum import IslandModeEnum
+
+logger = logging.getLogger(__name__)
+
+
+class IslandMode(flask_restful.Resource):
+    @jwt_required
+    def post(self):
+        body = json.loads(request.data)
+        mode_str = body.get("mode")
+        try:
+            mode = IslandModeEnum(mode_str)
+            set_mode(mode)
+            return make_response({}, 200)
+        except ValueError:
+            return make_response({}, 404)
```

Comment:
404 means that the URL was not found. 400 would be a better return code. Better yet, it should return 200 with an error in the response body.

Reply:
I'd say changing this to 400 is more than enough for now. There's no point in the error message, since this is not gonna be displayed anywhere and 200 will not throw a console error, thus once we add more modes, if the mode is miss-spelled on the UI it will be harder to notice

Reply:
I like 422